common/djangoapps/static_replace/test/test_static_replace.py

```diff
@@ -228,6 +231,24 @@ def setUpClass(cls):
                 cls.create_arbitrary_content(prefix, 'special/{}_excluded.html')
 
     @classmethod
+    def get_content_digest_for_asset_path(cls, prefix, path):
+        """
+        Takes an unprocessed asset path, parses it just enough to try and find the
+        asset it refers to, and returns the content digest of that asset if it exists.
+        """
+
+        # Parse the path as if it was potentially a relative URL with query parameters,
+        # or an absolute URL, etc.  Only keep the path because that's all we need.
+        _, _, relative_path, _, _, _ = urlparse(path)
+        asset_key = StaticContent.get_asset_key_from_path(cls.courses[prefix].id, relative_path)
+
+        try:
+            content = AssetManager.find(asset_key, as_stream=True)
```

Comment:
I think it's cheap enough if we're instructing browsers and CDNs to cache unlocked course assets for a very long time.  Unless the MongoDB driver, or MongoDB itself, are prefetching the file contents or something.. I would imagine opening in stream mode nets us just a fetch of the metadata and not the content itself, which should be pretty darn fast.